public/app/features/plugins/partials/ds_http_settings.html

```diff
@@ -94,6 +95,17 @@ <h6>Basic Auth Details</h6>
   </div>
 </div>
 
+<!--oauth-->
+<div class="gf-form-group" ng-if="current.jsonData.custumHeaders">
+  <h6>Custom Headers (for multiple headers split with ",")</h6>
```

Comment:
think this should be an array with an action button "Add custom header" (so current.jsonData.customHeaders is an array)

Reply:
ok.

Reply:
This is a tricky area since secureJsonData expect string as value.
we can change it to interface{} but it means more code change.

using JsonData could be another option, but required more tweaks and code changes in diffrent  areas.

any thoughts ?

@torkelo @bergquist

Reply:
No that should not be required as secureJsonData is a map (key value with string values) so just put header key names in jsonData.header1 jsonData.header2 and values in secureJsonData.header1 etc

Reply:
Using `jsonData` for header names and `securejsonData` for values sounds like a good idea. I suggest you prefix the header key and value with something to make it easier to loop over them.

Reply:
yep, i'm about to use ng-repeater to iterate them.

Reply:
can you suggest how to retrive the $scope ds_edit_ctrl.

I wanted to implement somthing similar to this concept
http://jsfiddle.net/yhv9bjrx/2/

Reply:
@mrsiano Try adding a functions in https://github.com/grafana/grafana/blob/master/public/app/features/plugins/ds_edit_ctrl.ts for returning the data.